src/util/xeval/builtin_cast.rs

```diff
@@ -0,0 +1,337 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::i64;
+use tipb::expression::Expr;
+use util::codec::datum::Datum;
+use util::codec::mysql::{Decimal, Time, Duration, DEFAULT_FSP};
+use util::codec::convert;
+use super::{Evaluator, EvalContext, Result, Error};
+
+const ERROR_UNIMPLEMENTED: &'static str = "unimplemented";
+const TYPE_INT: &'static str = "int";
+
+fn invalid_type_error(datum: &Datum, expected_type: &str) -> Result<Datum> {
+    Err(Error::Eval(format!("invalid expr type: {:?}, expect: {}", datum, expected_type)))
+}
+
+fn check_datum_int(datum: &Datum) -> Result<()> {
+    match *datum {
+        Datum::I64(_) | Datum::U64(_) => Ok(()),
+        _ => Err(Error::Eval(format!("invalid expr type: {:?}, expect to be int", datum))),
+    }
+}
+
+impl Evaluator {
+    pub fn cast_int_as_int(&mut self, ctx: &EvalContext, expr: &Expr) -> Result<Datum> {
+        let child = try!(self.get_one_child(expr));
+        let d = try!(self.eval(ctx, child));
+        match d {
+            Datum::I64(_) => Ok(d),
+            Datum::U64(u) => {
+                let i = try!(convert::convert_uint_to_int(u, i64::MAX));
+                Ok(Datum::I64(i))
+            }
+            _ => invalid_type_error(&d, TYPE_INT),
+        }
+    }
+
+    pub fn cast_int_as_real(&mut self, ctx: &EvalContext, expr: &Expr) -> Result<Datum> {
+        let child = try!(self.get_one_child(expr));
+        let d = try!(self.eval(ctx, child));
+        match d {
+            Datum::I64(i) => Ok(Datum::F64(i as f64)),
+            Datum::U64(u) => Ok(Datum::F64(u as f64)),
+            _ => invalid_type_error(&d, TYPE_INT),
+        }
+    }
+
+    pub fn cast_int_as_string(&mut self, ctx: &EvalContext, expr: &Expr) -> Result<Datum> {
+        let child = try!(self.get_one_child(expr));
+        let d = try!(self.eval(ctx, child));
+        match d {
+            Datum::I64(_) | Datum::U64(_) => {}
+            _ => return invalid_type_error(&d, TYPE_INT),
+        }
+        let s = try!(d.into_string());
+        Ok(Datum::Bytes(s.into_bytes()))
+    }
+
+    pub fn cast_int_as_decimal(&mut self, ctx: &EvalContext, expr: &Expr) -> Result<Datum> {
+        let child = try!(self.get_one_child(expr));
+        let datum = try!(self.eval(ctx, child));
+        let decimal = match datum {
+            Datum::I64(_) => Decimal::from(datum.i64()),
+            Datum::U64(_) => Decimal::from(datum.u64()),
+            _ => return invalid_type_error(&datum, TYPE_INT),
+        };
+        Ok(Datum::Dec(decimal))
+    }
+
+    pub fn cast_int_as_time(&mut self, ctx: &EvalContext, expr: &Expr) -> Result<Datum> {
+        let child = try!(self.get_one_child(expr));
+        let d = try!(self.eval(ctx, child));
+        try!(check_datum_int(&d));
+        let s = try!(d.into_string());
+        let t = try!(Time::parse_datetime(&s, DEFAULT_FSP, &ctx.tz));
```

Comment:
Yes. You could check it out in
https://github.com/pingcap/tikv/blob/master/src/util/codec/mysql/mod.rs#L25